app/src/main/java/org/simple/clinic/drugs/selection/custom/CustomDrugEntrySheet.kt

```diff
@@ -148,8 +148,11 @@ class CustomDrugEntrySheet : BaseBottomSheet<
     drugFrequencyEditText.setText(getString(frequencyLabelRes))
   }
 
-  override fun setSheetTitle(sheetTitle: String) {
-    titleTextView.text = sheetTitle
+  override fun setSheetTitle(drugName: String?, dosage: String?, frequencyLabelResID: Int) {
+    val hasDrugFrequency = frequencyLabelResID != R.string.custom_drug_entry_sheet_frequency_none
+    val frequency = if (hasDrugFrequency) getString(frequencyLabelResID) else null
+
+    titleTextView.text = listOfNotNull(drugName, dosage, frequency).joinToString()
```

Comment:
Is there a possibility here of all the elements being `null` and the title text being set to an empty string?

Reply:
No.. In no case we will have an empty drug name. Dosage and frequency can be null. The name can not be edited as well. It can only be edited while searching a string from drug search screen.